app/assets/javascripts/discourse/app/models/login-method.js

```diff
@@ -41,6 +41,11 @@ export default class LoginMethod extends EmberObject {
     return this.title_override || i18n(`login.${this.name}.title`);
   }
 
+  @discourseComputed
+  icon() {
+    return this.icon_override || i18n(`login.${this.name}.icon`);
```

Comment:
```suggestion
    return this.icon_override || "user";
```

We should probably default to "user" here instead of "i18n(login.${this.name}.icon`)" because when we output this it's in an icon component, right?

Reply:
For title and pretty_name there are actually two ways to override. You can set an i18n string `login.{auth_type}.title` or `login.{auth_type}.pretty_name`. Those can be overridden in admin > site texts, which is nice. Or you can pass a pretty_name_setting/title_setting when registering with `auth_provider`. The site setting takes precident.

With the way I have this set up now, the same applies to icon. So in the SAML plugin (or another) we could set a default icon value in client.en.yml _or_ with a site setting using `icon_setting`. I guess the yml route is more consistent with what other auth plugins are using (and the existing way I override the login "with SAML" text etc).

There is still a fallback to the "user" icon in the UI, but that could be moved here.

Reply:
The `i18n` thing makes sense for title/name, since they might need to be translated on multilingual sites. I don't think that's true for the icon. So I'd lean towards just keeping it as a static value or a site setting.

Reply:
Ok, that makes sense. I removed the i18n option, SiteSetting only now.